DOCS-159: Refactor PEM server and agent registration (#5409)
EnterpriseDB/docs: product_docs/docs/pem/8/registering_agent.mdx

- /pem/latest/pem_inst_guide_linux/04_installing_postgres_enterprise_manager/09_registering_a_pem_agent/ 
---

Before a PEM agent can be used, you must register it with a PEM server. 
**PEM agents installed by the PEM server package are registered automatically during server configuration.** For all other agents you must follow the instructions below.

!!! Note

>     After upgrading the PEM agent, you need to restart it. You do not need to register it again. 

Reviewer: ```suggestion
    After upgrading the PEM agent, you need to restart it. You don't need to register it again. 
```

## How to register PEM agents

> On Linux and Windows hosts, the PEM agent package includes command line utility called 'pemworker' which you can use perform management tasks, including registering the PEM agent, as described [below](#registering-a-pem-agent-using-the-pemworker-utility).

Reviewer: ```suggestion
On Linux and Windows hosts, the PEM agent package includes a command line utility called `pemworker`, which you can use to perform management tasks, including registering the PEM agent, as described [below](#registering-a-pem-agent-using-the-pemworker-utility).
```

> On Windows, the PEM agent graphical installer allows you to register the agent when installing it. This is a convenience option and does not support all the options provided by pemworker. 

Reviewer: ```suggestion
On Windows, the PEM agent graphical installer allows you to register the agent when installing it. This convenience option doesn't support all the possibilities provided by the pemworker. 
```

> If you do not want the installer to register the agent, uncheck the 'register now' checkbox. For more details, refer to the [installation instructions](/pem/latest/installing/windows/).

Reviewer: ```suggestion
If you don't want the installer to register the agent, uncheck the `register now` checkbox. For more details, refer to the [installation instructions](/pem/latest/installing/windows/).
```

## Registering a PEM agent using the pemworker utility

> The PEM agent package installer places the pemworker utility in the `/usr/edb/pem/agent/bin` directory on Linux and `C:\Program Files\edb\pem\agent-x64`bin` on Windows. 

Reviewer: ```suggestion
The PEM agent package installer places the pemworker utility in the `/usr/edb/pem/agent/bin` directory on Linux and `C:\Program Files\edb\pem\agent-x64\bin` on Windows. 
```

> To register an agent, invoke the utility as shown below and add the relevant options from the table as needed. Follow each option with a corresponding value.

Reviewer: ```suggestion
To register an agent, invoke the utility below and add the relevant options from the table as needed. Follow each option with a corresponding value.
```

Author: The utility isn't below, an invocation of it is. Therefore, I think the original is more precise.

| `--efm-install-path`                | Specifies the complete path to the installation directory of EDB Failover Manager (if applicable). Optional.                                                                                                                                                                                                                                                                                                |
| `--config-dir`                      | Specifies the directory path of the agent configuration file. Optional, defaults to `<pemworker path>/../etc`.                                                                                                                                                                                                                                                                                              |

> If you want to use any PEM feature for which a database server restart is required by the PEM agent (such as Audit Manager, Log Manager, or the Tuning Wizard), then you must set the value for `allow_server_restart` to `true` in the `agent.cfg` file.

Reviewer: ```suggestion
If you use any feature of PEM that requires a database server restart by the PEM agent (such as Audit Manager, Log Manager, or the Tuning Wizard), then you must set the value for `allow_server_restart` to `true` in the `agent.cfg` file.
```

!!! Note
    After upgrading the PEM agent, you need to restart it. You don't need to register it again. 

## How to register PEM agents
On Linux and Windows hosts, the PEM agent package includes a command line utility called `pemworker`, which you can use to perform management tasks, including registering the PEM agent, as described [below](#registering-a-pem-agent-using-the-pemworker-utility).

On Windows, the PEM agent graphical installer allows you to register the agent when installing it. This convenience option doesn't support all the possibilities provided by the pemworker. 
If you don't want the installer to register the agent, uncheck the `register now` checkbox. For more details, refer to the [installation instructions](/pem/latest/installing/windows/).

## Registering a PEM agent using the pemworker utility
The PEM agent package installer places the pemworker utility in the `/usr/edb/pem/agent/bin` directory on Linux and `C:\Program Files\edb\pem\agent-x64\bin` on Windows. 
To register an agent, invoke the utility as shown below and add the relevant options from the table as needed. Follow each option with a corresponding value.

#### Linux

```shell

> # Running as root

Reviewer: ```suggestion
# Running as root
export PEM_SERVER_PASSWORD=edb
```

- /pem/latest/pem_inst_guide_linux/04_installing_postgres_enterprise_manager/09_registering_a_pem_agent/ 
---

Before a PEM agent can be used, you must register it with a PEM server. 
**PEM agents installed by the PEM server package are registered automatically during server configuration.** For all other agents you must follow the instructions below.

!!! Note
    After upgrading the PEM agent, you need to restart it. You don't need to register it again. 

## How to register PEM agents
On Linux and Windows hosts, the PEM agent package includes a command line utility called `pemworker`, which you can use to perform management tasks, including registering the PEM agent, as described [below](#registering-a-pem-agent-using-the-pemworker-utility).

On Windows, the PEM agent graphical installer allows you to register the agent when installing it. This convenience option doesn't support all the possibilities provided by the pemworker. 
If you don't want the installer to register the agent, uncheck the `register now` checkbox. For more details, refer to the [installation instructions](/pem/latest/installing/windows/).

## Registering a PEM agent using the pemworker utility

> The PEM agent package installer places the pemworker utility in the `/usr/edb/pem/agent/bin` directory on Linux and `C:\Program Files\edb\pem\agent-x64\bin` on Windows. 

Author: This probably comes from the days of the linux installers. I'll change.
